### PR TITLE
Use map for pivot

### PR DESCRIPTION
**Pull request by tzufgoogle**
1. Change all features extracted from the map (pivots, intersections and points) to work with the map instead of extracting them from API (Slow).
2. Found bugs in the map - > fixed (distance calculation incorrect).
3. made some more changes in the map to support the feature extraction: for edge of type poi, the osmid changed to be as the osmid of the street the point was projected on.
4. Changed tests to support the changes (1-3).

### CODE REVIEW

File: cabby/geo/map_processing/edge.py [hunk 1]
```diff
@@ -55,13 +53,13 @@ def from_projected(cls, u_for_edge, v_for_edge, length, highway, osmid, name):
     )
 
   @classmethod
-  def from_poi(cls, u_for_edge, v_for_edge, osmid):
+  def from_poi(cls, u_for_edge, v_for_edge, osmid, dist):
```
Comment (jpalowitch) [inline cabby/geo/map_processing/edge.py:56]: nit: probably best to add Arguments, Returns to the docstring here

File: cabby/geo/map_processing/map_structure.py [hunk 2]
```diff
@@ -39,7 +39,8 @@
 # This variable is used (:map_structure.py; :edge.py; cabby.geo.util.py,
 # cabby.geo.walk.py) to project the geometries into this CRS for geo operations
 # such as calculating the centroid.
-OSM_CRS = 32633
+OSM_CRS = 4326
+POI_DISTANCE = 50
```
Comment (jpalowitch) [inline cabby/geo/map_processing/map_structure.py:43]: Please comment this module-level variable as well.

File: cabby/geo/regions.py [hunk 3]
```diff
@@ -32,8 +34,15 @@ def get_region(region: Text):
...
             ccw=True)
-  else: # D.C
-    polygon_area = box(
-        miny=38.90821, minx=-77.04053, maxy=38.90922, maxx=-77.03937,
-        ccw=True)
-  return polygon_area  
\ No newline at end of file
+  elif region == "DC": # D.C
+    polygon_area = box(minx=-77.02767, 
+                      miny=38.96608,
```
Comment (jpalowitch) [inline cabby/geo/regions.py:39]: nit: this argument and the two below should be indented 1 space

File: cabby/geo/regions.py [hunk 4]
```diff
@@ -32,8 +34,15 @@ def get_region(region: Text):
     polygon_area = box(
             miny=40.425, minx=-80.035, maxy=40.460, maxx=-79.930,
             ccw=True)
-  else: # D.C
-    polygon_area = box(
-        miny=38.90821, minx=-77.04053, maxy=38.90922, maxx=-77.03937,
-        ccw=True)
-  return polygon_area  
\ No newline at end of file
+  elif region == "DC": # D.C
```
Comment (jpalowitch) [inline cabby/geo/regions.py:37]: Where is the `if` parent to this `elif`?

File: cabby/geo/util.py [hunk 5]
```diff
@@ -345,26 +347,72 @@ def check_if_geometry_in_polygon(geometry: Any, poly: Polygon) -> Polygon:
...
+def get_polygon_distance_from_point(poly: Polygon, point: Point) -> int:
+  '''Calculate the distance between point and polygon in meters.
+  Arguments:
+    route: The line that length calculation will be performed on.
+    point: point to measure distance from polygon.
+  Returns:
+    The distance between point and polygon in meters.
+  '''
+  dist = 0
+  point_1 =  Point(poly.exterior.coords[0])
```
Comment (jpalowitch) [inline cabby/geo/util.py:372]: nit: remove one space after `=`

File: cabby/geo/walk.py [hunk 6]
```diff
@@ -36,10 +36,49 @@
 
 _Geo_DataFrame_Driver = "GPKG"
 OSM_CRS = 32633  # UTM Zones (North).
+SEED = 4
```
Comment (jpalowitch) [inline cabby/geo/walk.py:39]: Document this and OSM_CRS?

File: cabby/geo/walk.py [hunk 7]
```diff
@@ -36,10 +36,49 @@
...
+POI_DISTANCE = 50
```
Comment (jpalowitch) [inline cabby/geo/walk.py:40]: Looks like two files use this module-level variable. Could it go in a module that both of them import? (if that is inconvenient, the way you have it is probably fine)

File: cabby/geo/walk.py [hunk 8]
```diff
@@ -36,10 +36,49 @@
...
 
-def compute_route(start_point: Point, end_point: Point, graph: nx.MultiDiGraph,
-          nodes: GeoDataFrame) -> Optional[GeoDataFrame]:
+def compute_route_from_nodes(origin_id: int, 
+                              goal_id: int, 
+                              graph: nx.MultiDiGraph,
+                              nodes: GeoDataFrame) -> Optional[GeoDataFrame]:
+  '''Returns the shortest path between a starting and end point.
+  Arguments:
+  origin_id: The node id of the origin point.
```
Comment (jpalowitch) [inline cabby/geo/walk.py:49]: Nit: indent the list of Args and Returns

File: cabby/geo/walk.py [hunk 9]
```diff
@@ -90,7 +129,7 @@ def get_end_poi(map: map_structure.Map) -> Optional[GeoSeries]:
     return None
 
   # Pick random POI.
-  poi = small_poi.sample(1, random_state=1).iloc[0]
+  poi = small_poi.sample(1, random_state=SEED).iloc[0]
```
Comment (jpalowitch) [inline cabby/geo/walk.py:132]: How many functions use `SEED`? If not many, I would suggest making it a function argument, and then adding it as a flag to py_binaries.

File: cabby/geo/walk.py [hunk 10]
```diff
@@ -107,23 +146,19 @@ def get_start_poi(
...
-
   # Get closest nodes to points.
-  dest = ox.get_nearest_node(
-    map.nx_graph, util.tuple_from_point(end_point['centroid']))
+  dest_osmid = end_point['osmid']
 
   # Find nodes whithin 2000 meter path distance.
   outer_circle_graph = ox.truncate.truncate_graph_dist(
-    map.nx_graph, dest, max_dist=2000, weight='length')
+    map.nx_graph, dest_osmid, max_dist=2000+POI_DISTANCE, weight='length')
```
Comment (jpalowitch) [inline cabby/geo/walk.py:154]: nit: spaces around `+`, here and below

File: cabby/geo/walk.py [hunk 11]
```diff
@@ -107,23 +146,19 @@ def get_start_poi(
...
 
   outer_circle_graph_osmid = list(outer_circle_graph.nodes.keys())
 
   try:
-    # Get graph that is too close (less than 400 meter path distance)
+    # Get graph that is too close (less than 200 meter path distance)
     inner_circle_graph = ox.truncate.truncate_graph_dist(
-      map.nx_graph, dest, max_dist=400, weight='length')
+      map.nx_graph, dest_osmid, max_dist=200+POI_DISTANCE, weight='length')
```
Comment (jpalowitch) [inline cabby/geo/walk.py:161]: Should the number `200` be configurable?

File: cabby/geo/walk.py [hunk 12]
```diff
@@ -208,13 +250,12 @@ def get_pivot_near_goal(map: map_structure.Map, end_point: GeoSeries) -> \
   A single landmark near the goal location.
   '''
 
-  tags = {'name': True, 'wikidata': True,
-      'amenity': True, 'shop': True, 'tourism': True}
-  try:
-    poi = ox.pois.pois_from_point(util.tuple_from_point(
-      end_point['centroid']), tags=tags, dist=40)
+  near_poi_con = map.poi.apply(lambda x: util.get_distance_between_geometries(
+    x.geometry,end_point['centroid'])<80, axis=1)
```
Comment (jpalowitch) [inline cabby/geo/walk.py:254]: nit: spaces around `<` and `,`

File: cabby/geo/walk.py [hunk 13]
```diff
@@ -432,36 +463,27 @@ def get_number_intersections_past(main_pivot: GeoSeries, route: GeoDataFrame,
...
-  pivot_goal_route = route.loc[pivot_idx:]
-
-  if pivot_goal_route.shape[0] <= 1:
-    return -1
 
-  # Check if main pivot is in the pivot_goal_route segment.
-  segment_pivot = LineString(pivot_goal_route['geometry'].iloc[0:2].values)
-  number_intersection = -util.project_point_in_segment(
-    segment_pivot, main_pivot['geometry'])
+  pivot_goal_route = compute_route_from_nodes(main_pivot['osmid'],
```
Comment (jpalowitch) [inline cabby/geo/walk.py:467]: nit: Bring the first arg down a line, indent 4 spaces, and bring the other args back to 4 space indent as well

File: cabby/geo/map_processing/edge.py (hunk 1 above)
Comment (tzufgoogle) [reply cabby/geo/map_processing/edge.py:56]: Added.

File: cabby/geo/map_processing/map_structure.py (hunk 2 above)
Comment (tzufgoogle) [reply cabby/geo/map_processing/map_structure.py:43]: removed. This isn't needed after all

File: cabby/geo/regions.py (hunk 3 above)
Comment (tzufgoogle) [reply cabby/geo/regions.py:39]: fixed

File: cabby/geo/regions.py (hunk 4 above)
Comment (tzufgoogle) [reply cabby/geo/regions.py:37]: line 29  if region == "Manhattan":

File: cabby/geo/util.py (hunk 5 above)
Comment (tzufgoogle) [reply cabby/geo/util.py:372]: fixed

File: cabby/geo/walk.py (hunk 7 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:40]: this is removed

File: cabby/geo/walk.py (hunk 6 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:39]: removed - not used

File: cabby/geo/walk.py (hunk 8 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:49]: fixed indentation

File: cabby/geo/walk.py (hunk 9 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:132]: I made a change there in another PR so I'm going to leave it for now as it is.

File: cabby/geo/walk.py (hunk 10 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:154]: removed

File: cabby/geo/walk.py (hunk 11 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:161]: Changed to MAX_PATH_DIST and MIN_PATH_DIST

File: cabby/geo/walk.py (hunk 12 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:254]: fixed

File: cabby/geo/walk.py (hunk 13 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:467]: changed

**Review by jpalowitch — APPROVED** [review]

**Pull request merged**